nautobot/dcim/api/serializers.py

```diff
@@ -173,25 +173,34 @@ class PathEndpointModelSerializerMixin(ValidatedModelSerializer):
 
     @extend_schema_field(serializers.CharField(allow_null=True))
     def get_connected_endpoint_type(self, obj):
-        if obj._path is not None and obj._path.destination is not None:
-            return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
+        try:
+            if obj._path is not None and obj._path.destination is not None:
+                return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
+        except CablePath.DoesNotExist:
+            pass
```

Comment:
Consider (as @timizuoebideri1 has been doing in some of his recent code changes) using [contextlib.suppress](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) for these, I think that would be for example:

```suggestion
        with contextlib.suppress(CablePath.DoesNotExist):
            if obj._path is not None and obj._path.destination is not None:
                return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
```